### PR TITLE
miri-unleash tests: ensure they fire even with 'allow(const_err)'

### PR DESCRIPTION
**Pull request by RalfJung**
This is easier with `static` than `const` so I switched some of them over.

### CODE REVIEW

File: src/test/ui/consts/miri_unleashed/const_refers_to_static.rs [hunk 1]
```diff
@@ -1,39 +1,37 @@
...
-    //~^ WARN skipping const checks
-};
-fn main() {}
+fn main() {
+    MUTATE_INTERIOR_MUT;
+    //~^ ERROR: erroneous constant used
+    READ_INTERIOR_MUT;
+    //~^ ERROR: erroneous constant used
+    READ_MUT;
+    //~^ ERROR: erroneous constant used
```
Comment (RalfJung) [inline src/test/ui/consts/miri_unleashed/const_refers_to_static.rs:36]: Sometimes the error is "erroneous constant used", other times it is "evaluation of constant expression failed". I am not sure where the inconsistency is coming from.

File: src/test/ui/consts/miri_unleashed/const_refers_to_static.rs [hunk 2]
```diff
@@ -1,39 +1,37 @@
...
 // compile-flags: -Zunleash-the-miri-inside-of-you
-#![warn(const_err)]
+#![allow(const_err)]
 
 #![feature(const_raw_ptr_deref)]
 
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering;
 
-const REF_INTERIOR_MUT: &usize = { //~ ERROR undefined behavior to use this value
```
Comment (oli-obk) [inline src/test/ui/consts/miri_unleashed/const_refers_to_static.rs:9]: why did you remove this one?

Comment (oli-obk) [reply src/test/ui/consts/miri_unleashed/const_refers_to_static.rs:9]: oh, nevermind, you made a separate test

File: src/test/ui/consts/miri_unleashed/mutable_const.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 #![feature(const_raw_ptr_deref)]
 #![feature(const_mut_refs)]
-#![deny(const_err)]
+#![deny(const_err)] // FIXME: ICEs with allow! See #71316.
```
Comment (oli-obk) [inline src/test/ui/consts/miri_unleashed/mutable_const.rs:5]: yea, as you noted on the issue, I expect some things to ICE with unleash, and I think that's perfectly ok.

Comment (oli-obk) [reply src/test/ui/consts/miri_unleashed/mutable_const.rs:5]: can you add the appropriate compiletest comments to this file and allow the lint?

Comment (RalfJung) [reply src/test/ui/consts/miri_unleashed/mutable_const.rs:5]: I don't think it's okay that they ICE, but we can discuss that in the issue. ;)

I'll try to make the test check for an ICE instead.

Comment (RalfJung) [reply src/test/ui/consts/miri_unleashed/mutable_const.rs:5]: Oh, actually we already have a test for exactly that, `mutable_const2.rs`.

Comment (RalfJung) [reply src/test/ui/consts/miri_unleashed/mutable_const.rs]: @oli-obk I think my latest commit addresses... oh you already saw it and `r+`'d, never mind. ;)

File: src/test/ui/consts/miri_unleashed/mutable_const.rs [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
 const MUTABLE_BEHIND_RAW: *mut i32 = &UnsafeCell::new(42) as *const _ as *mut _;
 //~^ WARN: skipping const checks
 
-const MUTATING_BEHIND_RAW: () = {
+const MUTATING_BEHIND_RAW: () = { //~ NOTE
     // Test that `MUTABLE_BEHIND_RAW` is actually immutable, by doing this at const time.
     unsafe {
         *MUTABLE_BEHIND_RAW = 99 //~ ERROR any use of this value will cause an error
+        //~^ NOTE: which is read-only
+        // FIXME would be good to match more of the error message here, but looks like we
```
Comment (RalfJung) [inline src/test/ui/consts/miri_unleashed/mutable_const.rs:23]: Is there no way to normalize the messages that `NOTE`/`WARN`/`ERROR` match against? It did not seem to work here.

Comment (oli-obk) [reply src/test/ui/consts/miri_unleashed/mutable_const.rs:23]: nope, these can't be normalized

**Comment by RalfJung** [comment]
Rebased.
@bors r=oli-obk

**Comment by RalfJung** [comment]
GHA PR builders always "cancel" themselves it seems -- Azure builders are fine (Cc @pietroalbini)
